examples/cpp/custom_payload_pool/SubscriberApp.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file SubscriberApp.cpp
+ *
+ */
+
+#include "SubscriberApp.hpp"
+
+#include <condition_variable>
+#include <csignal>
+
+#include <fastdds/dds/core/status/SubscriptionMatchedStatus.hpp>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
+#include <fastdds/dds/subscriber/qos/SubscriberQos.hpp>
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+
+using namespace eprosima::fastdds::dds;
+namespace eprosima {
+namespace fastdds {
+namespace examples {
+namespace custom_payload_pool {
+
+SubscriberApp::SubscriberApp(
+        const CLIParser::subscriber_config& config,
+        const std::string& topic_name)
+    : participant_(nullptr)
+    , subscriber_(nullptr)
+    , topic_(nullptr)
+    , reader_(nullptr)
+    , type_(new CustomPayloadPoolDataPubSubType())
+    , samples_(config.samples)
+    , received_samples_(0)
+    , stop_(false)
+{
+    payload_pool_ = std::make_shared<CustomPayloadPool>();
+
+    // Create the participant
+    auto factory = DomainParticipantFactory::get_instance();
+    participant_ = factory->create_participant_with_default_profile(nullptr, StatusMask::none());
```

Comment:
Consider applying the same changes here as in the publisher (regarding new argument `--domain`) and not using profiles